packages/app/src/providers/utils.ts

```diff
@@ -124,8 +124,8 @@ export function createBasicFetcher(
 
       throw new FetcherError(response.status, response.statusText, buffer);
     } catch (error) {
-      if (error instanceof DOMException && error.name === 'AbortError') {
-        throw new AbortError(abortSignal, error);
+      if (abortSignal?.aborted && abortSignal.reason instanceof AbortError) {
+        throw abortSignal.reason;
```

Comment:
As explained, we have access to the `abortSignal`, so we can just check for `abortSignal.aborted`!

The `instanceof` check is just in case the signal is aborted in an unexpected way (i.e. not by the suspense fetch store).